mlir/lib/CAPI/IR/BuiltinAttributes.cpp

```diff
@@ -770,6 +770,34 @@ const void *mlirDenseElementsAttrGetRawData(MlirAttribute attr) {
 // Resource blob attributes.
 //===----------------------------------------------------------------------===//
 
+bool mlirAttributeIsADenseResourceElements(MlirAttribute attr) {
+  return llvm::isa<DenseResourceElementsAttr>(unwrap(attr));
+}
+
+MlirAttribute mlirUnmanagedDenseResourceElementsAttrGet(
+    MlirType shapedType, MlirStringRef name, const void *data,
+    size_t dataLength,
```

Comment:
if changing this anyways, might want to add `size_t align` and `bool dataIsMutable` as args for completeness. (and at that point I'm not actually sure what the point of going through `UnmanagedAsmResourceBlob` is, since it's just directly forwarding arguments to the `AsmResourceBlob` constructor?)

Reply:
Oh, good call. That flushed out a bug: the alignment was being inferred as 1. I've made this explicit and in the Python bindings, where we have strides, used that to infer the alignment of the data if not specified.